src/state/pages/actions.js

```diff
@@ -132,51 +131,49 @@ export const clearTree = () => ({
   type: CLEAR_TREE,
 });
 
-// TODO: generalize and centralize this function to cleanup API calls
 const wrapApiCall = apiFunc => (...args) => async (dispatch) => {
   const response = await apiFunc(...args);
-  const contentType = response.headers.get('content-type');
-  if (contentType && contentType.includes('application/json')) {
-    const json = await response.json();
-    if (response.ok) {
-      return json;
-    }
-    dispatch(addErrors(json.errors.map(e => e.message)));
-    throw json;
+  const json = await response.json();
+  if (response.ok) {
+    return json;
   }
-  throw new TypeError('No JSON content-type in response headers');
+  dispatch(addErrors(json.errors.map(e => e.message)));
+  throw json;
 };
 
 
 export const fetchPage = wrapApiCall(getPage);
 export const fetchPageChildren = wrapApiCall(getPageChildren);
 
 export const sendDeletePage = page => async (dispatch) => {
-  const response = await deletePage(page);
-  const contentType = response.headers.get('content-type');
-  if (contentType && contentType.includes('application/json')) {
+  try {
+    const response = await deletePage(page);
     const json = await response.json();
     if (response.ok) {
       dispatch(removePage(page));
       gotoRoute(ROUTE_PAGE_TREE);
-      return json;
+    } else {
+      dispatch(addErrors(json.errors.map(e => e.message)));
     }
-    dispatch(addErrors(json.errors.map(e => e.message)));
-    throw json;
+  } catch (e) {
+    // do nothing
   }
-  throw new TypeError('No JSON content-type in response headers');
 };
 
 export const fetchPageTree = pageCode => async (dispatch) => {
-  if (pageCode === HOMEPAGE_CODE) {
-    const responses = await Promise.all([
-      fetchPage(pageCode)(dispatch),
-      fetchPageChildren(pageCode)(dispatch),
-    ]);
-    return [responses[0].payload].concat(responses[1].payload);
+  try {
+    if (pageCode === HOMEPAGE_CODE) {
+      const responses = await Promise.all([
+        fetchPage(pageCode)(dispatch),
+        fetchPageChildren(pageCode)(dispatch),
+      ]);
+      return [responses[0].payload].concat(responses[1].payload);
+    }
+    const response = await fetchPageChildren(pageCode)(dispatch);
+    return response.payload;
+  } catch (e) {
+    throw e;
```

Comment:
No point in catch -> throw, you can get rid of the try/catch block here and it will be equivalent and more readable